python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -590,20 +597,9 @@ def get_eligible_operators(
     eligible_ops: List[PhysicalOperator] = []
 
     for op, state in topology.items():
-        assert resource_manager.op_resource_allocator_enabled(), topology
-
-        # Check whether the operator is under its limits imposed by the
-        # resource manager
-        under_resource_limits = (
-            resource_manager.op_resource_allocator.can_submit_new_task(op)
-        )
-        # Operator is considered being in task-submission back-pressure if
-        # both of the following holds true:
-        #   - It's exceeding its resource limits
-        #   - At least one of the back-pressure policies are violated
-        in_backpressure = not under_resource_limits or not all(
-            p.can_add_input(op) for p in backpressure_policies
-        )
+        # Operator is considered being in task-submission back-pressure if any
+        # back-pressure policy is violated
+        in_backpressure = any(not p.can_add_input(op) for p in backpressure_policies)
```

Comment:
As @omatthew98 pointed out above -- let's make back-pressuring traceable (by logging when the op becomes throttled first time)

Reply:
The op will be flipping between backpressure and non-backpressure status pretty frequently. I don't think logging the first time would be useful.